pkg/reconciler/service/service.go

```diff
@@ -148,11 +148,13 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 		return err
 	}
 
+	rc := service.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady)
 	if config.Generation != config.Status.ObservedGeneration {
 		// The Configuration hasn't yet reconciled our latest changes to
 		// its desired state, so its conditions are outdated.
 		service.Status.MarkConfigurationNotReconciled()
-	} else {
+	}
+	if config.Generation == config.Status.ObservedGeneration || (rc != nil && rc.Status == corev1.ConditionFalse) {
```

Comment:
I suspect the bug here is that Configuration / Route aren't updating the ObservedGeneration for which they are failing.

Reply:
+1

This condition will be more clear if we instead make sure that we bump "ObservedGeneration" in the Route and Configuration reconcilers  when "Ready" is updated to "False".

Reply:
+1
For this case, "ObservedGeneration" in the configuration reconciler is already updated.
In router reconciler, "ObeservedGeneration" will not be updated until route is successfully synced. 
Implementation in Route and Configuration reconcilers is a little different.